cql3/query_processor.cc

```diff
@@ -1056,9 +1056,17 @@ query_processor::execute_schema_statement(const statements::schema_altering_stat
             co_await stmt.grant_permissions_to_creator(client_state);
         }
     }
-
-    // If an IF [NOT] EXISTS clause was used, this may not result in an actual schema change.  To avoid doing
-    // extra work in the drivers to handle schema changes, we return an empty message in this case. (CASSANDRA-7600)
+    if (!ce) {
+        // If an IF [NOT] EXISTS clause was used and resource was already created
+        // we don't emit schema change event. However it interacts badly with
```

Comment:
In this patch, we do emit the event, no?

Now I understand the virtual function.

Reply:
yes

Reply:
Then the wording is confusing, the comment says we don't emit the event, the code says we do.

Reply:
should be "shouldn't" instead of "don't"

Reply:
fixed